docs/miscellaneous.rst

```diff
@@ -58,6 +61,51 @@ So for the following contract snippet::
 
 The position of ``data[4][9].b`` is at ``keccak256(uint256(9) . keccak256(uint256(4) . uint256(1))) + 1``.
 
+Adding elements to or removing them from a dynamically-sized array will increment or decrement the number
+of elements which is stored in the slot. For arrays that are not of type ``bytes`` or ``string`` this is
+straight-forward, since data is stored in a different slot.
+
+
+``bytes`` and ``string``
+------------------------
+
+``bytes`` and ``string`` store their data in the same slot where also the length is stored if they are short. In
```

Comment:
Perhaps start with stating that bytes and string are encoded identically.

Reply:
Also for clarification, perhaps start with `for short byte arrays, ...`

Reply:
I must admit I also find this quite confusing, the original was more complex, but maybe clearer. What was the reason to change it? Was it wrong?